### PR TITLE
Add pagination and refactor services into classes

### PR DESCRIPTION
**Pull request by kharann**
## Changelog

- More fluent way to paginate lists of data
- Fixed broken vitests
- Converted companies to the new class service format

### CODE REVIEW

File: apps/dashboard/src/app/event/EventCreationModal.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ export type EventCreationModalProps = {
 }
 
 export const EventCreationModal: FC<EventCreationModalProps> = ({ close }) => {
-  const { data: committees = [] } = trpc.committee.all.useQuery({ limit: 50 })
+  const { data: committees = [] } = trpc.event.all.useQuery()
```
Comment (matsjla) [inline apps/dashboard/src/app/event/EventCreationModal.tsx:13]: Should be trpc.committee.all

File: packages/api/src/utils/db-utils.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    cursor: CursorSchema.optional(),
+  })
+  .optional()
+  .default({ take: 20, cursor: undefined })
+
+export type Cursor = z.infer<typeof CursorSchema>
+
+export const paginateQuery = (qb: AnySelectQueryBuilder, cursor: Cursor) => {
+  return qb
+    .where((qb) => qb.where("createdAt", "=", cursor.createdAt).where("id", "<", cursor.id))
```
Comment (matsjla) [inline packages/api/src/utils/db-utils.ts:21]: How does this work? Doesn't this become uuid comparison?

**Review by matsjla — CHANGES_REQUESTED** [review]

File: packages/api/src/utils/db-utils.ts (hunk 2 above)
Comment (kharann) [reply packages/api/src/utils/db-utils.ts:21]: I just did this https://dba.stackexchange.com/questions/267794/how-to-do-pagination-with-uuid-v4-and-created-time-on-concurrent-inserted-data

Comment (kharann) [reply packages/api/src/utils/db-utils.ts:21]: image

**Review by matsjla — APPROVED** [review]
epic

**Pull request merged**